packages/admin/admin/src/FinalForm.tsx

```diff
@@ -177,12 +176,15 @@ export function FinalForm<FormValues = AnyObject>(props: IProps<FormValues>) {
         );
     }
 
-    function handleSubmit(values: FormValues, form: FormApi<FormValues>) {
+    async function handleSubmit(values: FormValues, form: FormApi<FormValues>) {
+        editDialogFormApi?.onFormStatusChange("saving");
+
+        await waitForFormValidationToBeFinished(form);
```

Comment:
This ensures that the update mutation doesn't overtake the validation. This can make the submission take a bit longer, but prevents cases where the update fails and the validation error isn't shown yet (-> bad UX). 
This would probably happen very rarely in practice since the update mutation should also validate all fields.

Reply:
final-form awaits async validations before submitting: https://github.com/final-form/final-form/blob/main/src/FinalForm.js#L1128

Now the question is why it doesn't in our case?

Reply:
Ah, I understand now.

> In my case, it doesn't because I don't call FinalForms submit() anymore. I call our handleSubmit() function directly, so I have to await the validations myself.

why that? That doesn't sound like something you should do?

Reply:
> why that? That doesn't sound like something you should do?

I believe we changed that in #1414. @thomasdax98 can you confirm?